### ISSUE TITLE
Pico W tls_client tls_client_err -14

### ISSUE THREAD

**Issue by GoldeggCoder** [raspberrypi/pico-examples#573]
Recently, I attempted to transmit an HTTPS request using the Pico W. I cloned the example „tls_client“ and attempted to build it. This process was successful, but upon running the request on the Pico W, I encountered an error message indicating „tls_client_err -14“ after successfully resolving the domain. I am currently utilizing the Pico SDK version 2.0.

**Comment by Potzi123** [comment]
I have the same Problem pls help

**Comment by peterharperuk** [comment]
ERR_RST: the connection was reset by the remote host

**Comment by GoldeggCoder** [comment]
But I am using the basic example with the domain provided.(worldtimeapi) Why does they close the connection?

**Comment by peterharperuk** [comment]
It's not our website, but it appears to be down. We need to change it to something else.

**Comment by GoldeggCoder** [comment]
Yeah thanks for the help. I got it working for my own domain. However it is very slow. It takes about 20 seconds to send a request. (See #411) Any fix for that?

**Comment by lurch** [comment]
https://github.com/micropython/micropython/issues/16233 suggests that many sites refuse to respond to requests without a User-Agent header, so maybe https://github.com/raspberrypi/pico-examples/blob/master/pico_w/wifi/tls_client/picow_tls_client.c#L11 ought to include a `User-Agent` header too?

**Comment by peterharperuk** [comment]
> Yeah thanks for the help. I got it working for my own domain. However it is very slow. It takes about 20 seconds to send a request. (See https://github.com/raspberrypi/pico-examples/issues/411) Any fix for that?

Make sure you're using a release build!
